react/components/loaders/FetchMoreButton.js

```diff
@@ -1,36 +1,25 @@
-import React from 'react'
+import React, { Fragment } from 'react'
 import { Button } from 'vtex.styleguide'
 import { FormattedMessage } from 'react-intl'
 
-import SearchResult from '../SearchResult'
-import { loaderPropTypes } from '../../constants/propTypes'
-
 import searchResult from '../../searchResult.css'
 
-/**
- * Search Result Component.
- */
-const ShowMoreLoaderResult = props => {
+const FetchMoreButton = props => {
   const {
     products,
+    to,
     recordsFiltered,
     onFetchMore,
     fetchMoreLoading,
     showProductsCount,
-    children,
   } = props
 
-  // If it has children, it is a flexible UI
-  if (children) {
-    return children
-  }
-
   return (
-    <SearchResult {...props}>
+    <Fragment>
       <div
         className={`${searchResult.buttonShowMore} w-100 flex justify-center`}
       >
-        {!!products && products.length < recordsFiltered && (
+        {!!products && to + 1 < recordsFiltered && (
```

Comment:
not sure if I'm mistaken here, but
```suggestion
        {!!products && to <= recordsFiltered && (
```